batavia/types/List.js

```diff
@@ -72,14 +71,29 @@ batavia.types.List = function() {
     List.prototype.__lt__ = function(other) {
 
 
-        if(batavia.isinstance(other, [batavia.types.Bytes, batavia.types.Bytearray])){
+        if (batavia.isinstance(other, [batavia.types.Bytes, batavia.types.Bytearray])){
             throw new batavia.builtins.TypeError("unorderable types: list() < " + batavia.type_name(other) + "()")
         }
 
         if (other !== null) {
             if (batavia.isinstance(other, batavia.types.List)) {
-                /* update this line to get Pythonic list < list behavior */
-                return this.valueOf() < other;
+                // edge case where this==[]
+                if (this.length === 0 && other.length > 0){
+                    return true;
+                }
+
+                for (var i=0; i<this.length; i++){
+
+                    //other ran out of items.
+                    if (other[i] === undefined){
+                        return false;
+                    }
+                    if (this[i].__ne__(other[i])){
+                        return this[i].__lt__(other[i]);
+                    }
+                }
+                //got through loop and all values were equal. Determine by comparing length
+                return this.length < other.length;
```

Comment:
Thanks for the input! I'm a little confused about comparing lengths first. My understanding of the behavior was that we compare the first item of both lists, then the second and so on. When a non equal elements are found, their result is the result of the comparison. Comparing just the length of two lists doesn't alone tell us how they compare, we need to compare elements one by one. Example:

```
>>> [3,4] > [1,2,3]
True
```
What am I missing?

source: https://docs.python.org/3/tutorial/datastructures.html#comparing-sequences-and-other-types